architecture-guide.md

```diff
@@ -0,0 +1,357 @@
+
+# Status Desktop Architecture Guide
```

Comment:
This is only about the UI architecture, maybe we coudl rename it to such so we leave room to add a Nim side architecture and good practices guide.

Reply:
Yeah, I will rename it according to @igor-sirotin's suggestion.